packages/react-scripts/config/webpack.config.js

```diff
@@ -750,29 +753,31 @@ module.exports = function (webpackEnv) {
           // The formatter is invoked directly in WebpackDevServerUtils during development
           formatter: isEnvProduction ? typescriptFormatter : undefined,
         }),
-      new ESLintPlugin({
-        // Plugin options
-        extensions: ['js', 'mjs', 'jsx', 'ts', 'tsx'],
-        formatter: require.resolve('react-dev-utils/eslintFormatter'),
-        eslintPath: require.resolve('eslint'),
-        context: paths.appSrc,
-        cache: true,
-        cacheLocation: path.resolve(
-          paths.appNodeModules,
-          '.cache/.eslintcache'
-        ),
-        // ESLint class options
-        cwd: paths.appPath,
-        resolvePluginsRelativeTo: __dirname,
-        baseConfig: {
-          extends: [require.resolve('eslint-config-react-app/base')],
-          rules: {
-            ...(!hasJsxRuntime && {
-              'react/react-in-jsx-scope': 'error',
-            }),
+      !disableESLintPlugin &&
+        new ESLintPlugin({
+          // Plugin options
+          extensions: ['js', 'mjs', 'jsx', 'ts', 'tsx'],
+          formatter: require.resolve('react-dev-utils/eslintFormatter'),
+          eslintPath: require.resolve('eslint'),
+          emitWarning: isEnvDevelopment && emitErrorsAsWarnings,
```

Comment:
I think adding this `emitWarning` flag had the unintended effect of hiding warnings entirely if `ESLINT_NO_DEV_ERRORS` is not `true`. I was able to confirm that the warnings correctly appear if I set that env variable but then eslint errors are downgraded to warnings. I dug around the `eslint-webpack-plugin` codebase a bit and I think we might not even need to set this option since the default is true already.

There was also a bug fix recently in [`eslint-webpack-plugin@2.5.0`](https://github.com/webpack-contrib/eslint-webpack-plugin/releases/tag/v2.5.0) to fix an issue with `emitWarning` and `emitError` so `react-scripts` might also want to bump up that version requirement if this option is staying in.

Reply:
I read through the linked issue and I think the `failOnError` option was the one people wanted here - maybe something like this?

```suggestion
          ...(isEnvDevelopment && emitErrorsAsWarnings && { failOnError: false }),
```

Reply:
Hi @DarkAce65, are you able to provide a reproduction of this?

Our error logging won't show warnings when errors are present (as far as I remember), so could that be what you mean?

I can see locally that if I have only a warning, no errors, I see that warning. If I have errors and warnings, I see only errors.

Reply:
Hey @mrmckeb, that's odd, it's not what I was seeing. You're right that if there's an error, no warnings show up but I'm also not seeing any warnings if there are no errors :(

There's an example in facebook/create-react-app#10509 but I can try and make a reproduction repo in a few hours.

Out of curiosity, what version of `eslint-webpack-plugin` do you have locally? Also, you have `ESLINT_NO_DEV_ERRORS` set to false right?

Reply:
Thanks for the update, I have `eslint-webpack-plugin@2.4.3`.

If I run `npx create-react-app --info`, I see:

```
  Binaries:
    Node: 14.15.0 - ~/.nvm/versions/node/v14.15.0/bin/node
    Yarn: 1.22.5 - /usr/local/bin/yarn
    npm: 6.14.8 - ~/.nvm/versions/node/v14.15.0/bin/npm
  Browsers:
    Chrome: 88.0.4324.150
    Edge: Not Found
    Firefox: Not Found
    Safari: 14.0.3
  npmPackages:
    react: ^17.0.1 => 17.0.1 
    react-dom: ^17.0.1 => 17.0.1 
    react-scripts: 4.0.2 => 4.0.2 
  npmGlobalPackages:
    create-react-app: Not Found
```

Reply:
Did a bit more testing, seems like the combination of `react-scripts@4.0.2` and `eslint-webpack-plugin@2.5.0` causes the missing warnings: https://github.com/DarkAce65/cra-eslint-warnings